### DISCUSSION TITLE
[help] Testing WCF Services with Cypress

### DISCUSSION THREAD

**Discussion by vishallanke** [cypress-io/cypress#25142] [Questions and Help]
Can we test Microsoft WCF Services using Cypress ?
WCF Services are hosted with various endpoints like BasicHttpBinding, NetTCP, NetMSMQ et. - https://www.c-sharpcorner.com/UploadFile/0c1bb2/endpoints-in-wcf/

**Comment by accessinstitute** [comment]
I would not recommend testing of WCF services with Cypress. We have tried this in our project and messed up test pipelines. The best way to test WCF service is using Dot Net. WCF should not be treated as API for test automation using Cypress. You can test simple GET WCF Calls using Cypress. But for complex POST/PUT calls where you have to send objects by doing some conversion, serialization, compression, payload modification etc. it is not recommended to use Cypress or in that case even Selenium. 

The best way is to create test project in Dot Net, develop some unit test cases their and test WCF Calls.

```
export function getDetails(){
  return cy.request({
      method: "GET",
      url: `https://myproject.com/myproject.svc`,
      headers: {
          "Content-Type": "text/xml; charset=utf-8",
          "SOAPAction": `http://tempuri.org/IUserService/GetDetails`    
      },
      body: null
  })
  .then(function (response) {
      return response;
  });
}  



Format of Body...just an example

  let soapRequest =`<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:tem="http://tempuri.org/">
  <soapenv:Header/>
  <soapenv:Body>
     <tem:getDetails>
        <tem:guid>ahd713673-12388912388-123jnsadfhwerud-12438712347</tem:guid>
     </tem:getDetails>
  </soapenv:Body>
</soapenv:Envelope>`;

```


Hope it helps !

**Comment by DeLaphante** [comment]
Struggling to design or build an advanced, stable and well structured test automation framework?

Check out & play around with my powerful and robust Selenium Web UI & API test automation framework on Github -> https://github.com/DeLaphante/CynkyAutomation

Save yourself some time and use this ready made framework for any project and get the technical support you need to succeed!
